dbms/src/Storages/KVStore/Utils/AsyncTasks.h

```diff
@@ -24,13 +24,27 @@ template <typename Key, typename Func, typename R>
 struct AsyncTasks
 {
     // We use a big queue to cache, to reduce add task failures.
-    explicit AsyncTasks(uint64_t pool_size)
-        : thread_pool(std::make_unique<ThreadPool>(pool_size, pool_size, 300))
-    {}
     explicit AsyncTasks(uint64_t pool_size, uint64_t free_pool_size, uint64_t queue_size)
         : thread_pool(std::make_unique<ThreadPool>(pool_size, free_pool_size, queue_size))
     {}
 
+    bool discardTask(Key k)
+    {
+        std::scoped_lock l(mtx);
+        auto it = futures.find(k);
+        if (it != futures.end())
+        {
+            futures.erase(it);
+            auto it2 = start_time.find(k);
+            if (it2 != start_time.end())
+            {
+                start_time.erase(it2);
+            }
```

Comment:
```suggestion
            start_time.erase(it2);
```
These lines can be simplified to a single `erase` call because there is no else branch

Reply:
fixed